compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -69,9 +69,41 @@ object Objects:
 
   // ----------------------------- abstract domain -----------------------------
 
+  /** Syntax for the data structure abstraction used in abstract domain:
+   *
+   * ve ::= ObjectRef(class)
+   *      | OfClass(class, vs[outer], ctor, args, env)
+   *      | OfArray(object[owner], regions)
+   *      | Fun(..., env)                                                // value elements that can be contained in ValueSet
+   * vs ::= ValueSet(ve)                                                 // set of abstract values
+   * Bottom ::= ValueSet(Empty)
+   * val ::= ve | Cold | vs                                              // all possible abstract values in domain
+   * Ref ::= ObjectRef | OfClass                                         // values that represents a reference
+   * ThisValue ::= Ref | Cold                                            // possible values for 'this'
+   *
+   * refMap = Ref -> ( valsMap, varsMap, outersMap )                     // refMap stores field informations of an object or instance
+   * valsMap = valsym -> val                                             // maps immutable fields to their values
+   * varsMap = valsym -> addr                                            // each mutable field has an abstract address
+   * outersMap = class -> val                                            // maps outer objects to their values
```

Comment:
As we have discussed, `outer` could be any possible value in our current implementation (see `outerValue`)